presto-main/src/main/java/com/facebook/presto/sql/Optimizer.java

```diff
@@ -160,7 +160,7 @@ private StatsAndCosts computeStats(PlanNode root, TypeProvider types)
                         (node instanceof JoinNode) || (node instanceof SemiJoinNode)).matches()) {
             StatsProvider statsProvider = new CachingStatsProvider(statsCalculator, session, types);
             CostProvider costProvider = new CachingCostProvider(costCalculator, statsProvider, Optional.empty(), session);
-            return StatsAndCosts.create(root, statsProvider, costProvider);
+            return StatsAndCosts.create(root, statsProvider, costProvider, session, true);
```

Comment:
Should this be properly gated with a session property in case it goes wrong?

Reply:
Sure, let me add one.

Reply:
Added a session property for it